openmsupply/open-msupply
# 2067 contact trace frontend

Part of #2019

The contact trace UI works very similar to the encounter UI with the difference that a contact trace is created in the DetailView (encounter is created when scheduled and then edited in the DetailView).

Currently, a contact trace can be created without being enrolled in the matching program. For example, a contact trace might be created for a patient enrolled in a Testing OR a Care program, for this reason there is no restriction right now when a contact trace can be created (@richardmoizeau).

- Adds an "Add Contact Trace" option to the add button in the patient view
- This button leads to a modal that allows you to select a tracing type 
- The modal leads to an DetailView that can be filled to create the contact trace 
- The same DetailView is used to edit a contact trace
- There is a new Contact Traces tab with a list of contact traces for a single patient
- There is also a list in the left sidebar showing all contact traces for all patients

![image](https://github.com/openmsupply/open-msupply/assets/88299195/3fabf15d-88d2-4a04-b920-d175a5258538)

# 🧪 How has/should this change been tested? 


Please use the latest dev init script from the schema repository.
- Create a contact trace
- Edit a contact trace
- Check both contact trace lists
